contracts/cw-blotto/src/contract.rs

```diff
@@ -364,26 +376,49 @@ impl BlottoContract<'_> {
             )?;
         }
 
-        // TODO handle tie
-        // Determine over all winner
-        let game_winner = self
+        // Determine overall winner
+        // Start with list of armies
+        let armies = self
             .armies
             .range(ctx.deps.storage, None, None, Order::Descending)
             .map(|res| res.map(|(_, army)| army))
-            .collect::<Result<Vec<_>, _>>()?
-            .into_iter()
+            .collect::<Result<Vec<_>, _>>()?;
+        println!("armies: {:?}", armies);
+
+        // Sort it by number of victory points
+        let game_winner = armies
+            .iter()
             .max_by_key(|army| army.victory_points)
             .ok_or(ContractError::InvalidGameWinnerDetermination {})?;
-        self.winner.save(ctx.deps.storage, &game_winner)?;
-
-        // Save the prize pool amount for the winning army
-        self.prize_pool.save(ctx.deps.storage, &prize_pool)?;
+        println!("game winner: {:?}", game_winner);
+
+        // Check on duplicated max points
+        let is_duplicated = armies
+            .iter()
+            .filter(|army| army.victory_points == game_winner.victory_points)
+            .count()
+            > 1;
+        println!("is duplicated: {}", is_duplicated);
```

Comment:
This is fine, but we're doing a lot of iteration. Might be some ways to make it more efficient. :wink: